### PR TITLE
Fix for issue 254

### PR DESCRIPTION
**Pull request by ctalledo**

### CODE REVIEW

File: libsysbox/syscont/spec.go [hunk 1]
```diff
@@ -690,24 +692,68 @@ func sysMgrSetupMounts(mgr *sysbox.Mgr, spec *specs.Spec, uidShiftRootfs bool) e
...
+			// If the special dir exists within the container's rootfs, then uid
+			// shifting is required when it's owned by root:root.
+
+			st, _ := fi.Sys().(*syscall.Stat_t)
+
+			if st.Uid != st.Gid {
+				return fmt.Errorf("container rootfs has special dir %s with non-matching uid & gid: %u %u", path, st.Uid, st.Gid)
+			}
+
+			if st.Uid != 0 && st.Uid != uid {
```
Comment (rodnymolina) [inline libsysbox/syscont/spec.go:723]: Don't we need this same exact if-statement for st.Gid?

File: libsysbox/syscont/spec.go [hunk 2]
```diff
@@ -690,24 +692,68 @@ func sysMgrSetupMounts(mgr *sysbox.Mgr, spec *specs.Spec, uidShiftRootfs bool) e
...
+				shiftUids = true
+			}
+
+		} else if os.IsNotExist(err) {
+
+			// If the special dir does not exist within the container rootfs, then
+			// uid shifting is not required when the container starts. However,
+			// when the container stops, if the sysbox host volume that backs the
+			// mount over the special dir has data in it, we copy that back to the
+			// container's rootfs. In this case, whether we do uid shifting on the
```
Comment (rodnymolina) [inline libsysbox/syscont/spec.go:737]: Looks like you're missing a noun here: "pid shifting on the *** during ...".

**Review by rodnymolina — APPROVED** [review]
LG. Take a look at my two comments.

File: libsysbox/syscont/spec.go (hunk 1 above)
Comment (ctalledo) [reply libsysbox/syscont/spec.go:723]: Yes you are right; I added it.

File: libsysbox/syscont/spec.go (hunk 2 above)
Comment (ctalledo) [reply libsysbox/syscont/spec.go:737]: Yep, fixed it.

**Pull request merged**